oracle/src/sponge.rs

```diff
@@ -149,55 +149,57 @@ where
         self.last_squeezed = vec![];
         for g in g.iter()
         {
-            if g.infinity {
-                panic!("marlin sponge got zero curve point");
-            } else {
-                self.sponge.absorb(&self.params, &[g.x]);
-                self.sponge.absorb(&self.params, &[g.y]);
-            }
+            self.sponge.absorb(&self.params, &[g.x]);
```

Comment:
Shouldn't we still panic here? The x and y coordinates are not valid for points which are infinity.

Reply:
If we cannot panic, we would like to have some kind of sentinel affine value for points at infinity, like (0, 0). Then we can replicate that behavior in the circuit.

Reply:
These comments have been addressed in commit 861b022a66a331a41407b79882295ca86b18a464 by restoring the original logic of Rust runtime panic in case of the absorption of the Point at Infinity.